source/wait_time.c

```diff
@@ -0,0 +1,27 @@
+#ifndef STRUCTS
+#define STRUCTS
+
+#include "../headers/structs.h"
+
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+
+/*Checking if passengers can finish*/
+void wait_time(transition_system *t) {
+    int i, j , p_W;
+    passenger p;
+    for(i = 0; i < t->length; i++) {
+        p = t->passengers[i];
+        p_W = p.wait_time;
+        for (j = 0; j < t->passengers[i].wait_time ; j++){
+            if (p_W > 0) {
+                p_W -= 1;
```

Comment:
I stedet for at ændre på to variabler, kan I bare modificere den korrekte variabel direkte

```suggestion
                t->passengers[i].wait_time -= 1;
```